[docs] Update mui-x on material-ui navigation (#31810)
mui/material-ui: docs/data/material/components/pickers/deprecated-pickers.md

> # Date / Time pickers

Reviewer: Is there a reason for capitalizing Time here? The headline seems to be referring to the general category of these components, as opposed to `<DatePicker>` and `<TimePicker>` themselves. 

Also, I think it looks a lot cleaner to ditch the "/" in favor of "and."

```suggestion
# Date and time pickers
```

Reviewer 2: I'd suppose it's because you can also have `<DateTimePicker>`. But I personally think it's way cleaner the way you're proposing here.

> <p class="description">Now part of MUI X, a collection of advanced components.</p>

Reviewer: Let's make this a complete sentence, and name the components that have been moved. Since it looks like there are quite a few, it may be best to just give some examples at least.

```suggestion
<p class="description">The `<DatePicker>`, `<TimePicker>`, and related components have been moved to MUI X—our collection of advanced components.</p>
```

> ## What is the reason to move the Pickers to MUI X?

Reviewer: Grammar. Also, I don't think "Picker" should be capitalized since we're talking about a category rather than a specific component.

```suggestion
## Why have the picker components been moved to MUI X?
```

> MUI X components are those meant to support complex use cases. Pickers can seem trivial but they're considerably tricky to pull off correctly, so we've moved them over to the X collection to make sure we're investing proper time and effort. If you want to know more details, check out the [blog post](/blog/lab-pickers-to-mui-x/).

Reviewer: Cut unnecessary wording, rephrase to be less "we"-centric, make the link text more descriptive.

```suggestion
MUI X components are built for complex use cases. Date and time pickers are difficult to pull off correctly, so they deserve the special attention that they will receive as part of MUI X. To learn more, check out the [blog post about the move](/blog/lab-pickers-to-mui-x/).
```

> ## How to migrate?

Reviewer: Grammar.

```suggestion
## How do I migrate?
```

Follow the [migration page](/x/react-date-pickers/migration/) to migrate from lab to MUI X.

## Where is the new documentation?

> You can find Date / Time pickers' documentation in the [MUI X docs](/x/react-date-pickers/).

Reviewer: Grammar and style.

```suggestion
You can find documentation on the date and time picker components in the [MUI X docs](/x/react-date-pickers/).
```

> ## How to report issues, request a feature, or submit a PR?

Reviewer: Grammar.

```suggestion
## Where should I report issues, request features, or submit PRs?
```

> You should go to the [MUI X repository](https://github.com/mui/mui-x) to do those things now that the Pickers have been moved there.

Reviewer: ```suggestion
Please share all feedback, issues, and requests in the [MUI X repository](https://github.com/mui/mui-x).
```

# Date / Time pickers

<p class="description">Now part of MUI X, a collection of advanced components.</p>

## What is the reason to move the Pickers to MUI X?

MUI X components are those meant to support complex use cases. Pickers can seem trivial but they're considerably tricky to pull off correctly, so we've moved them over to the X collection to make sure we're investing proper time and effort. If you want to know more details, check out the [blog post](/blog/lab-pickers-to-mui-x/).

## How to migrate?

Follow the [migration page](/x/react-date-pickers/migration/) to migrate from lab to MUI X.

## Where is the new documentation?

You can find Date / Time pickers' documentation in the [MUI X docs](/x/react-date-pickers/).

## How to report issues, request a feature, or submit a PR?

> Given the Pickers have been moved to MUI X, you should go to the [MUI X repository](https://github.com/mui/mui-x). to do those things.

Reviewer 2: ```suggestion
Given the Pickers have been moved to MUI X, you should go to the [MUI X repository](https://github.com/mui/mui-x) to do those things.
```

# Date / Time pickers (deprecated)

> <p class="description">This package has been moved to MUI X.</p>

Reviewer 2: ```suggestion
<p class="description">Now part of MUI X, a collection of advanced components.</p>
```

> ⚠️ The pickers in the lab are deprecated and have been moved to MUI X.

> ## What is the reason to move the pickers to MUI X?

Reviewer 2: ```suggestion
## What is the reason to move the Pickers to MUI X?
```

😅

It's become clear that the pickers are categorized as advanced components. By moving them to MUI X, we will be able to add/manage new features and include bug fixes more effectively. If you want to know more details about this afford, check out the [blog post](/blog/lab-pickers-to-mui-x/).

## How to migrate?

Follow the [migration page](/x/react-date-pickers/migration/) to migrate from lab to MUI X.

## Where is the new documentation?

> The [new documentation space](/x/react-date-pickers/) for date/time pickers is in MUI X.

Reviewer 2: ```suggestion
You can find Date / Time pickers' documentation in the [MUI X docs](/x/react-date-pickers/).
```

Though I wonder if I'm not being repetitive using docs twice...

> ## Where to report issues/request a feature or submit a PR?

Reviewer 2: ```suggestion
## How to report issues, request a feature, or submit a PR?
```
